File: sdk/@launchdarkly/mobile-dotnet/observability/replay/plugin/SessionReplayOptions.cs
using System;

namespace LaunchDarkly.SessionReplay;

public class SessionReplayOptions
{
    public class PrivacyOptions
    {
        public bool MaskTextInputs { get; set; } = true;
        public bool MaskWebViews { get; set; } = false;
        public bool MaskLabels { get; set; } = false;
        public bool MaskImages { get; set; } = false;
        public double MinimumAlpha { get; set; } = 0.02;

        public PrivacyOptions()
        {
        }

        public PrivacyOptions(
            bool maskTextInputs = true,
            bool maskWebViews = false,
            bool maskLabels = false,
            bool maskImages = false,
            double minimumAlpha = 0.02)
        {
            MaskTextInputs = maskTextInputs;
            MaskWebViews = maskWebViews;
            MaskLabels = maskLabels;
            MaskImages = maskImages;
            MinimumAlpha = minimumAlpha;
        }
    }

    public bool IsEnabled { get; set; } = true;
    public string ServiceName { get; set; } = "sessionreplay-dotnet";
    public PrivacyOptions Privacy { get; set; } = new PrivacyOptions();

    public SessionReplayOptions()
    {
    }

    public SessionReplayOptions(
        bool isEnabled = true,
        string serviceName = "sessionreplay-dotnet",
        PrivacyOptions? privacy = null
    )
    {
        IsEnabled = isEnabled;
        ServiceName = serviceName;
        Privacy = privacy ?? new PrivacyOptions();
    }
}

File: sdk/@launchdarkly/flutter/packages/observability/lib/src/session_replay_options.dart
// Ported from
// sdk/@launchdarkly/mobile-dotnet/observability/replay/plugin/SessionReplayOptions.cs.

import 'messages.g.dart' as wire;

/// Configuration for the LaunchDarkly session replay plugin. Mirrors
/// `LaunchDarkly.SessionReplay.SessionReplayOptions` in the .NET bridge.
class SessionReplayOptions {
  final bool isEnabled;
  final String serviceName;
  final PrivacyOptions privacy;

  const SessionReplayOptions({
    this.isEnabled = true,
    this.serviceName = 'sessionreplay-flutter',
    this.privacy = const PrivacyOptions(),
  });

  wire.LDSessionReplayOptions toWire() => wire.LDSessionReplayOptions(
    isEnabled: isEnabled,
    serviceName: serviceName,
    privacy: privacy.toWire(),
  );
}

/// Privacy controls for session replay. Mirrors
/// `SessionReplayOptions.PrivacyOptions` in the .NET bridge.
class PrivacyOptions {
  final bool maskTextInputs;
  final bool maskWebViews;
  final bool maskLabels;
  final bool maskImages;
  final double minimumAlpha;

  const PrivacyOptions({
    this.maskTextInputs = true,
    this.maskWebViews = false,
    this.maskLabels = false,
    this.maskImages = false,
    this.minimumAlpha = 0.02,
  });

  wire.LDPrivacyOptions toWire() => wire.LDPrivacyOptions(
    maskTextInputs: maskTextInputs,
    maskWebViews: maskWebViews,
    maskLabels: maskLabels,
    maskImages: maskImages,
    minimumAlpha: minimumAlpha,
  );
}
